impress-org/form-builder-library
# Feature: add ClassicEditor, OptionsPanel, and SettingsSection components

## Description



This PR adds the following component to the library:

1. The `ClassicEditor` component which uses the TinyMCE editor shipped with WordPress under the hood.
2. The `OptionsPanel` component that can provide multiple arrangements to attend a variety of settings necessary in different blocks.
3. The `SettingsSection` component that is used in the new settings UI to visually separate settings into horizontal sections.

## Affects



The following add-ons:

- FFM
- Tributes
- Funds

## Visuals



![image](https://github.com/impress-org/form-builder-library/assets/16308864/7cf08fbc-e61a-451e-9e45-c5038285affc)

![image](https://github.com/impress-org/form-builder-library/assets/16308864/8dfa19fc-87d9-4c42-a9e8-144bf06a671c)

## Testing Instructions



1. Pull this repository in the `wp-content/plugins` folder;
2. In the addon folder that you wish to test, go to the `package.json` file and add the following line as a dependency: `"@givewp/form-builder-library": "file:../form-builder-library/src"` and run the `npm install` command after that;
4. Go to the place in the addon codebase where one of these components is being used and replace the import line to use the component from the package installed in the previous step. 

For example, the following line: 

```
import ClassicEditor from "../ClassicEditor";
```

Should become something like this: 

```
import ClassicEditor from "@givewp/form-builder-library/ClassicEditor";
```

## Pre-review Checklist
